wnfs/src/private/hamt/node.rs

```diff
@@ -452,6 +458,187 @@ where
             Ok((Rc::new(node), removed))
         })
     }
+
+    /// Visits all the leaf nodes in the trie and calls the given function on each of them.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::rc::Rc;
+    /// use wnfs::{private::{Node, Pair}, utils, Hasher, MemoryBlockStore};
+    ///
+    /// #[async_std::main]
+    /// async fn main() {
+    ///     let store = &mut MemoryBlockStore::new();
+    ///     let mut node = Rc::new(Node::<[u8; 4], String>::default());
+    ///     for i in 0..99_u32 {
+    ///         node = node
+    ///             .set(i.to_le_bytes(), i.to_string(), store)
+    ///             .await
+    ///             .unwrap();
+    ///     }
+    ///
+    ///     let keys = node
+    ///         .flat_map(&|Pair { key, .. }| Ok(*key), store)
+    ///         .await
+    ///         .unwrap();
+    ///
+    ///     assert_eq!(keys.len(), 99);
+    /// }
+    /// ```
+    #[async_recursion(?Send)]
+    pub async fn flat_map<F, T, B>(&self, f: &F, store: &B) -> Result<Vec<T>>
```

Comment:
Should this perhaps be a stream instead?

I wish this could just be an `Iterator`, but yeah, it's async, so... `AsyncIterator` when? :P

Reply:
Intended to do this but forgot.